tools/twix/src/panels/enum_plot.rs

```diff
@@ -0,0 +1,354 @@
+use std::{
+    collections::hash_map::DefaultHasher,
+    hash::{Hash, Hasher},
+    str::FromStr,
+    sync::Arc,
+};
+
+use communication::client::CyclerOutput;
+use eframe::{
+    egui::{show_tooltip_at_pointer, ComboBox, Response, RichText, Ui, Widget},
+    epaint::{Color32, Stroke},
+};
+use egui_plot::{Plot, PlotBounds, PlotPoint, PlotUi, Polygon, Text};
+use itertools::Itertools;
+use log::{error, info};
+use serde_json::Value;
+
+use crate::{
+    change_buffer::{Change, ChangeBuffer},
+    completion_edit::CompletionEdit,
+    nao::Nao,
+    panel::Panel,
+};
+
+fn color_hash(value: impl Hash) -> Color32 {
+    let mut hasher = DefaultHasher::new();
+    value.hash(&mut hasher);
+
+    let hash = hasher.finish();
+
+    let [r, g, b, ..] = hash.to_le_bytes();
+
+    Color32::from_rgb(r, g, b)
+}
+
+#[derive(Clone)]
+struct Segment {
+    start: usize,
+    end: usize,
+    value: Value,
+}
+
+impl Segment {
+    fn name(&self) -> String {
+        match &self.value {
+            Value::Null => "<Null>".into(),
+            Value::Bool(value) => value.to_string(),
+            Value::Number(_) => "<Number>".into(),
+            Value::String(string) => string.clone(),
+            Value::Array(_) => "<Array>".into(),
+            Value::Object(map) => {
+                // There is only one key in the map.
```

Comment:
I understand the need for a comment here, but maybe move in the if block. At least how it is written suggests so